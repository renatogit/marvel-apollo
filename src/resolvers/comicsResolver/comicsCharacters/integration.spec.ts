const DataSourcesComicsCharactersAPI = require('@/dataSources/comics');
const ComicsCharactersResolver = require('.');
const ServerGetComicsCharacters = require('@/configs/test');
const QUERY_GET_COMICS_CHARACTERS = require('@query/comics/queryComicsCharacters');
const MOCK_GET_COMICS_CHARACTERS = require('@mock/comics/mockComicsCharacters');

describe('resolvers/comicsCharactersResolver/integration', () => {
	const comicsId = {comicsId: 'fake-code'};

	it('fetches comicsCharacters and generate snapshots', async () => {
		const {server} = await ServerGetComicsCharacters(
			DataSourcesComicsCharactersAPI,
			ComicsCharactersResolver,
			MOCK_GET_COMICS_CHARACTERS,
			comicsId
		);

		const {body} = await server.executeOperation({
			query: QUERY_GET_COMICS_CHARACTERS,
			variables: comicsId,
		});

		expect(body).toMatchSnapshot();
	});
});
