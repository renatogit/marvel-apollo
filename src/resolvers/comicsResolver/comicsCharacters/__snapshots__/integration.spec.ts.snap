// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/comicsCharactersResolver/integration fetches comicsCharacters and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "comicsCharacters": {
        "attributionHTML": "mock data",
        "attributionText": "mock data",
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 1,
          "limit": 20,
          "offset": 0,
          "results": [
            {
              "comics": {
                "available": 161,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "description": "mock data",
              "events": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "id": 1011010,
              "modified": "mock data",
              "name": "mock data",
              "resourceURI": "mock data",
              "series": {
                "available": 161,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "stories": {
                "available": 161,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                    "type": "mock data",
                  },
                ],
                "returned": 20,
              },
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
          ],
          "total": 1,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
